managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -67,28 +66,27 @@ public static OpReadEntry create(ManagedCursorImpl cursor, Position readPosition
         return op;
     }
 
-    void internalReadEntriesComplete(List<Entry> returnedEntries, Object ctx, Position lastPosition) {
+    private void internalReadEntriesComplete(List<Entry> returnedEntries) {
+        if (returnedEntries.isEmpty()) {
+            checkReadCompletion();
+            return;
+        }
         // Filter the returned entries for individual deleted messages
         int entriesCount = returnedEntries.size();
         long entriesSize = 0;
-        for (int i = 0; i < entriesCount; i++) {
-            entriesSize += returnedEntries.get(i).getLength();
+        for (final var returnedEntry : returnedEntries) {
+            entriesSize += returnedEntry.getLength();
         }
```

Comment:
It's a common pattern in Pulsar source code to use a for loop in this type of cases. The reason for this is that this avoids creating an Iterator instance. I would avoid making such changes, although it would be "cleaner" to change this.

Reply:
I can revert this change since it's not a necessary part of this part and keep the original commit id for `git blame`.

But I don't agree that it's a common pattern in Pulsar.

```
% find pulsar-broker/src/main -name "*.java" | xargs grep -n "for (.*:" | wc -l 
     343
```

Creating an Iterator (in range-for loop or other cases) is widely accepted in **Java world**, it's hardly a real concern.